### DISCUSSION TITLE
Crowd-funding to get vscode/language-server support?

### DISCUSSION THREAD

**Discussion by guettli** [cue-lang/cue#2358] [General]
I know, that I can't convince my co-workers to use CUE if there is no good vscode integration.

Maybe I am missing something, but AFAIK autocomplete is not availabe up to now. Please correct me if I am wrong.

Sorry, I am not able to help with writing code.

But I could give some money. And if serveral people give money,, too, then a developer could get payed to get
the language-server implementation done.

What do you think?

**Comment by kajogo777** [comment]
My team could volunteer to take on this project at https://stakpak.dev

We adopted CUE and heavily rely on it in our projects, and we think the language server is a crucial component to increase user adoption.

We are already working on a language server for CUE, but it would be even better to involve more people to work on a  single implementation, rather than having multiple incomplete implementations out there.

Who will work on it? 
@thethumbler and I mainly

**Reply by guettli** [reply]
Is your current language server for CUE available somewhere?

**Reply by kajogo777** [reply]
No, still in very early development

Child values auto-complete
![cue_autocomplete_1](https://user-images.githubusercontent.com/10531031/234008820-73b87b09-21ae-4d4b-b57d-329b2c0383cd.png)

Embedded values auto-complete
![cue_autocomplete_2](https://user-images.githubusercontent.com/10531031/234008836-f88ae55c-1237-49a9-a0bf-efb2104ab16b.png)

**Reply by kajogo777** [reply]
It's not yet in a ready-to-use state, but we can make the repo public if this will help

**Reply by jmgilman** [reply]
Hi @kajogo777, have you guys perhaps made any progress on this? I'm also trying to increase adoption and a VSCode plugin would really help! :)

**Comment by verdverm** [comment]
Existing issue: https://github.com/cue-lang/cue/issues/142

**Comment by moderation** [comment]
There is a couple of VSCode extension:
- https://marketplace.visualstudio.com/items?itemName=nickgo.cuelang
- https://marketplace.visualstudio.com/items?itemName=asdine.cue
- https://marketplace.visualstudio.com/items?itemName=jallen7usa.vscode-cue-fmt

And a CUE LSP from the folks at @dagger which has now been archived:
- https://github.com/dagger/cuelsp
